trin-execution/src/era/manager.rs

```diff
@@ -17,42 +17,59 @@ use super::{
 };
 
 pub struct EraManager {
+    current_block_number: u64,
     current_era: Option<ProcessedEra>,
     next_era: Option<JoinHandle<ProcessedEra>>,
     http_client: Client,
     era1_files: Vec<String>,
 }
 
 impl EraManager {
-    pub async fn new() -> anyhow::Result<Self> {
+    pub async fn new(starting_block: u64) -> anyhow::Result<Self> {
         let http_client: Client = Config::new()
             .add_header("Content-Type", "application/xml")
             .expect("to be able to add header")
             .try_into()?;
         let era1_files = get_shuffled_era1_files(&http_client).await?;
```

Comment:
> now we should probably use `get_era1_files` and keep it as hashmap. This would remove looking up the right url from below.

my concern is what if a new epoch is added, but I guess it doesn't matter since we will continue syncing from the running CL client anyways, so it doesn't matter if we have a few new era files or not